### PR TITLE
Docs for Custom Roles

### PR DESCRIPTION
**Pull request by steplercamp**
## Summary & Motivation
This is a draft of documentation for custom roles

## How I Tested These Changes
Built locally

### CODE REVIEW

File: docs/content/dagster-plus/account/managing-users/managing-user-roles-permissions.mdx [hunk 1]
```diff
@@ -14,32 +14,50 @@ In this guide, we'll cover how RBAC works in Dagster+, how to assign roles to us
...
+To edit a role:
+1. Click the edit button next to the role you want to edit
+2. Make your changes
+3. Save your changes
+4. You will see a confirmation dialog including the changes that have been made
 ---
 
-## Assigning user and team roles
+## Deleting custom roles
+Deleting custom roles is similar to editting. At the bottom of the edit dialog you'll see a button to "Delete role". Two things to note when deleting a role:
```
Comment (PedramNavid) [inline docs/content/dagster-plus/account/managing-users/managing-user-roles-permissions.mdx:51]:
```suggestion
Deleting custom roles is similar to editing. At the bottom of the edit dialog you'll see a button to "Delete role". Two things to note when deleting a role:
```

File: docs/content/dagster-plus/account/managing-users/managing-user-roles-permissions.mdx [hunk 2]
```diff
@@ -14,32 +14,50 @@ In this guide, we'll cover how RBAC works in Dagster+, how to assign roles to us
...
-Dagster+ uses a hierarchical model for RBAC, meaning that the most permissive roles include permissions from the roles beneath them. The following user roles are currently supported, in order from the **most** permissive to the **least** permissive:
-
-- Organization Admin
-- Admin
-- Editor
-- Launcher (Pro plans only)
-- Viewer
-
-For example, the **Admin** user role includes permissions specific to this role and all permissions in the **Editor**, **Launcher**, and **Viewer** user roles. Refer to the [User permissions reference](#user-permissions-reference) for the full list of user permissions in Dagster+.
+There are two key concepts for Dagster+ user roles.
```
Comment (PedramNavid) [inline docs/content/dagster-plus/account/managing-users/managing-user-roles-permissions.mdx:17]: I might rephrase this a little. First, I wouldn't include dates/newly introduced in docs as these docs will be around for a while. 

---

Dagster+ employs a flexible approach to user roles and permissions. This system is built on two fundamental concepts:

1. **Permission Scope**: Permissions are context-specific, falling into two main categories:
   - Organization-wide settings (for example, "Create Teams")
   - Deployment-specific actions (for example, "Launch and Cancel Backfills")

2. **Role Types**:  Dagster+ supports two types of roles:
   - **Default Roles**: Hierarchical roles, based on sensible defaults that come out of the box with Dagster.
   - **Custom Roles**: Allows you to define tailored roles with specific sets of permissions

This dual approach provides both the simplicity of predefined roles and the flexibility to create custom roles that precisely match your organization's needs.

Comment (steplercamp) [reply docs/content/dagster-plus/account/managing-users/managing-user-roles-permissions.mdx:17]: Is it weird that I read "based on sensible defaults that come out of the box with Dagster." in your voice? 

But serious this is super helpful, thank you, making the changes now!

**Comment by steplercamp** [comment]
Closing this out and moving here per Dish's guidance! 

https://github.com/dagster-io/dagster/pull/25380

**Pull request closed without merging**